lambda/test_script.py

```diff
@@ -0,0 +1,239 @@
+
+    mock_create_google_client.return_value = GoogleClientMockClass
+    mock_get_credentials_file.return_value = (
+        "./lambda/test_assetts/expected_credentials_file.json"
+    )
+    mock_get_scope.return_value = {"Parameter": {"Value": "test_scope_value"}}
+    mock_get_subject_email.return_value = {"Parameter": {"Value": "test_subject_email"}}
+
+    os.environ["CREDENTIALS"] = "credentials_param"
+    os.environ["SUBJECT"] = "subject_email_param"
+    os.environ["test_groups_scope"] = "test_groups_scope"
+
+    group_dict = {"group1": "group1@email.com", "group2": "group2@email.com"}
+
+    actual = get_group_info("http:api.com", "v3", "test_groups_scope", group_dict)
+    expected = [
+        {
+            "kind": "admin#directory#group",
+            "id": "id1",
+            "etag": "etag",
+            "email": "group1@email.com",
+            "name": "team1",
+            "directMembersCount": "0",
+            "description": "a description here",
+            "adminCreated": False,
+        },
+        {
+            "kind": "admin#directory#group",
+            "id": "id2",
+            "etag": "etag",
+            "email": "group2@email.com",
+            "name": "group2",
+            "directMembersCount": "3",
+            "description": "",
+            "adminCreated": False,
+        },
+    ]
+    assert expected == actual
+
+
+@patch("script.get_group_info")
+@patch("script.build_group_dict")
+def test_print_group_info(mock_build_group_dict, mock_get_group_info):
+    mock_build_group_dict.return_value = {
+        "group1": "group1@email.com",
+        "group2": "group2@email.com",
+    }
+    mock_get_group_info.return_value = [
+        {
+            "kind": "admin#directory#group",
+            "id": "id1",
+            "etag": "etag",
+            "email": "group1@email.com",
+            "name": "team1",
+            "directMembersCount": "0",
+            "description": "a description here",
+            "adminCreated": False,
+        },
+        {
+            "kind": "admin#directory#group",
+            "id": "id2",
+            "etag": "etag",
+            "email": "group2@email.com",
+            "name": "group2",
+            "directMembersCount": "3",
+            "description": "",
+            "adminCreated": False,
+        },
+    ]
+
+    capturedOutput = io.StringIO()
+    sys.stdout = capturedOutput
+    print_group_info(
+        "http://api.com", "v3", "scope", "http://adminapi.com", "v3", "admin scope"
+    )
+    sys.stdout = sys.__stdout__
+    actual = capturedOutput.getvalue()
+    expected = '{"kind": "admin#directory#group", "id": "id1", "etag": "etag", "email": "group1@email.com", "name": "team1", "directMembersCount": "0", "description": "a description here", "adminCreated": false}\n{"kind": "admin#directory#group", "id": "id2", "etag": "etag", "email": "group2@email.com", "name": "group2", "directMembersCount": "3", "description": "", "adminCreated": false}\n'  # noqa: E501
```

Comment:
I might be missing a trick, but can't you re-use the `mock_get_group_info.return_value` data with `json.dumps` to avoid having to write out this large line? It requires an extra import but it would make things clearer.

I've tested two options:

1. One liner using `str.join`, somewhat less obvious what's going on here
```python
    expected = (
        "\n".join((json.dumps(i) for i in mock_get_group_info.return_value)) + "\n"
    )
```

2. Multiple lines, running `json.dumps` on each item:
```python
    expected = (
        json.dumps(mock_get_group_info.return_value[0])
        + "\n"
        + json.dumps(mock_get_group_info.return_value[1])
        + "\n"
    )
```

(these are the outputs after running `make`, so they've been formatted via `black` already).

I haven't really understood the test and the rest of the scripts though, so I appreciate that I might be missing the point of this test by suggesting the above.

Reply:
oh nice suggestion! There was no real reason for that long line in the tests so I've swapped in your second suggestion here. Thanks!

Reply:
Late to the party - I also like Deniz's solutions, and agree that the 2nd suggestion is better for clarity.
My only tweak would be to use implicit string concatenation within brackets:
```python
expected = (
        json.dumps(mock_get_group_info.return_value[0])
        "\n"
        json.dumps(mock_get_group_info.return_value[1])
        "\n"
    )
```
But a tiny thing.